### PR TITLE
Feature/logic get all job title

### PR DESCRIPTION
**Pull request by Hasa9nAli**
Hi Dear Beautiful Reviewer this is our branch we 
1 - fix some bugs 
2-  add logic to GetAllJobTitleUseCase 
3 - add the Repository Provider
4 - add launch to test the data 
use fetch data from Api you need just call viewModel of Job in any Activity or fragment and watch Logcat then you will see amazing result 
thank you for you effort ( your colleagues @Hasa9nAli , @abuhussien28 )

### CODE REVIEW

File: app/src/main/java/com/cupcake/jobsfinder/data/repository/Repository.kt [hunk 1]
```diff
@@ -9,7 +9,7 @@ interface Repository {
 
   suspend fun getAllPosts(): Flow<List<PostDto>>
 
-  suspend fun getAllJobTitles():List<JobTitleDto>
+  suspend fun getAllJobTitles():Flow<List<JobTitleDto>>
```
Comment (AmeerAmjed) [inline app/src/main/java/com/cupcake/jobsfinder/data/repository/Repository.kt]: no need Flow is useless use **suspend** it's enough.

File: app/src/main/java/com/cupcake/jobsfinder/data/repository/RepositoryImpl.kt [hunk 2]
```diff
@@ -46,13 +46,15 @@ class RepositoryImpl @Inject constructor(
...
 	}
 
-	override suspend fun getAllJobTitles(): List<JobTitleDto> {
-		return listOf(
-			JobTitleDto(
-				id = "ID HERE",
-				title = "Android"
-			)
-		)
+	override suspend fun getAllJobTitles(): Flow<List<JobTitleDto>> {
```
Comment (AmeerAmjed) [inline app/src/main/java/com/cupcake/jobsfinder/data/repository/RepositoryImpl.kt]: this function need refactor

File: app/src/main/java/com/cupcake/jobsfinder/di/RepositoryModule.kt [hunk 3]
```diff
@@ -1,15 +1,28 @@
...
+import kotlinx.serialization.SerialInfo
+import javax.inject.Singleton
 
 @Module
 @InstallIn(SingletonComponent::class)
 abstract class RepositoryModule {
     @Binds
     abstract fun bindsRepository(repository: RepositoryImpl): Repository
+
+    companion object {
```
Comment (AmeerAmjed) [inline app/src/main/java/com/cupcake/jobsfinder/di/RepositoryModule.kt]: delete this function no need

File: app/src/main/java/com/cupcake/jobsfinder/domain/usecase/GetAllJobTitleUseCase.kt [hunk 4]
```diff
@@ -2,15 +2,15 @@ package com.cupcake.jobsfinder.domain.usecase
...
 import javax.inject.Inject
 
 class GetAllJobTitleUseCase @Inject constructor(
     private val jobTitleRepository: Repository,
 ) {
-    suspend operator fun invoke(): List<JobTitle> {
-        return jobTitleRepository.getAllJobTitles().map { jobTitleDto ->
-            jobTitleDto.toJobTitle()
-        }
+    suspend operator fun invoke()
```
Comment (AmeerAmjed) [inline app/src/main/java/com/cupcake/jobsfinder/domain/usecase/GetAllJobTitleUseCase.kt]: function neeed improve

**Review by AmeerAmjed — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/cupcake/jobsfinder/ui/jobs/JobsViewModel.kt [hunk 5]
```diff
@@ -15,7 +19,8 @@ import javax.inject.Inject
 
 @HiltViewModel
 class JobsViewModel @Inject constructor(
-    private val getAllJobUseCase: GetAllJobUseCase
+    private val getAllJobUseCase: GetAllJobUseCase,
+    private val repository: RepositoryImpl
```
Comment (TarekIdrees) [inline app/src/main/java/com/cupcake/jobsfinder/ui/jobs/JobsViewModel.kt]: Why did you inject repo impl in ViewModel ?

**Review by TarekIdrees — CHANGES_REQUESTED** [review]

**Review by msalarab — APPROVED** [review]
just change ameer's requests

File: app/src/main/java/com/cupcake/jobsfinder/data/repository/RepositoryImpl.kt (hunk 2 above)
Comment (AbuHussien28) [reply app/src/main/java/com/cupcake/jobsfinder/data/repository/RepositoryImpl.kt]: done

File: app/src/main/java/com/cupcake/jobsfinder/data/repository/Repository.kt (hunk 1 above)
Comment (AbuHussien28) [reply app/src/main/java/com/cupcake/jobsfinder/data/repository/Repository.kt]: done

File: app/src/main/java/com/cupcake/jobsfinder/di/RepositoryModule.kt (hunk 3 above)
Comment (AbuHussien28) [reply app/src/main/java/com/cupcake/jobsfinder/di/RepositoryModule.kt]: done

File: app/src/main/java/com/cupcake/jobsfinder/domain/usecase/GetAllJobTitleUseCase.kt (hunk 4 above)
Comment (AbuHussien28) [reply app/src/main/java/com/cupcake/jobsfinder/domain/usecase/GetAllJobTitleUseCase.kt]: done

File: app/src/main/java/com/cupcake/jobsfinder/ui/jobs/JobsViewModel.kt (hunk 5 above)
Comment (AbuHussien28) [reply app/src/main/java/com/cupcake/jobsfinder/ui/jobs/JobsViewModel.kt]: can for test and we forget to remove it 
now we remove it

**Review by AmeerAmjed — APPROVED** [review]
good job

**Pull request merged**